File: solvers.julia/toa_46/test_num_toa_46.jl
include("../_common/header.jl")
include("inidata_num_toa_46.jl")
include("coefs_toa_46.jl")
include("red_427x475_toa_46.jl")

function test_num_toa_46()

    Random.seed!(23)
    N = 10000
    d = 38

    stats = ini_stats("toa_46")

    for i in 1:N
        data = inidata_num_toa_46()
        C,U = coefs_toa_46(data)

        toc = @elapsed S = red_427x475_toa_46(C)
        stats.tm = [stats.tm; toc]
        if isempty(S); continue; end

        stats = update_stats(stats,S,C,U,d)
    end

    disp_stats(stats)

end

test_num_toa_46()

File: solvers.python/py_toa_46/test_num_toa_46.py
from numpy.random import seed
from time import time
from _common import ini_stats, update_stats, disp_stats
from py_toa_46 import inidata_num_toa_46, coefs_toa_46, red_427x475_toa_46

def test_num_toa_46():
    seed(23)
    N = 1000
    d = 38

    stats = ini_stats('toa_46')

    for i in range(N):
        data = inidata_num_toa_46()
        C,U = coefs_toa_46(data)

        tic = time()
        S = red_427x475_toa_46(C)
        stats['tm'].append(time()-tic)
        if S is None:
            continue

        stats = update_stats(stats,S,C,U,d)

    disp_stats(stats)